apps/hs/qua-server/src/Foundation.hs

```diff
@@ -482,3 +487,13 @@ lastScenarioProblemId = getVal <$> rawSql query []
     query = unlines
           ["SELECT max(id) FROM scenario_problem;"
           ]
+
+
+maybeEnroll :: Email -> Maybe Text -> Handler ()
+maybeEnroll email msId =
+    case msId of
+        Nothing -> pure ()
+        Just i -> do
+             $(logDebug) $ "Enrolling new user with email " <> email <> " in scenario problem " <> i
+             runDB $ pure () -- TODO what here?
```

Comment:
Here are the handlers that setup exercise in case a user have all session parameters:
[Handler.Mooc.EditProposal#L30](https://github.com/achirkin/qua-kit/blob/master/apps/hs/qua-server/src/Handler/Mooc/EditProposal.hs#L30),
[Handler.Mooc.CompareProposals#L33](https://github.com/achirkin/qua-kit/blob/master/apps/hs/qua-server/src/Handler/Mooc/CompareProposals.hs#L33)

So, the most important parameter is `userSessionCustomExerciseId` - id of scenario (exercise) they are working on.

Secondly, there is `userSessionCustomExerciseType` - either "edit" or "compare". For workshops default would be "edit".

Setting up the above two session parameters (both in user session and in DB) should be enough to "enroll" the student.

Lastly, I am not sure about edx-compliant params: `userSessionContextId` (edx course id), `userSessionResourceLink` (link to edx course block), `userSessionOutcomeServiceUrl` (edx link to send grade to), `userSessionResultSourceId` (id to specify in edX grading request). We can either keep them empty, or set some default stub value indicating that this is not an edx exercise. What do you think?